contracts/swaps/connectors/SwapsImplBalancer_POLYGON.sol

```diff
@@ -165,11 +172,10 @@ contract SwapsImplBalancer_POLYGON is State, ISwapsImpl {
             (IBalancerVault.BatchSwapStep[] memory swapParams, address[] memory tokens, int256[] memory limits) = abi.decode(payload, (IBalancerVault.BatchSwapStep[], address[], int256[]));
             require(tokens[0] == sourceTokenAddress && tokens[tokens.length-1] == destTokenAddress, "invalid tokens");
```

Comment:
this require and   _checkLimits(limits) can be moved above the condition, they are the same in both cases

Reply:
_checkLimits has loop inside and there is more and less the same loop right after _checkLimits call, maybe it's possible to combine and loop only once

Reply:
this has been adjusted